controllers/machinedeletionremediation_controller.go

```diff
@@ -133,9 +133,14 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 	}()
 
 	if r.isTimedOutByNHC(mdr) {
-		log.Info("NHC time out annotation found, stopping remediation")
-		commonevents.RemediationStoppedByNHC(r.Recorder, mdr)
-		_, err = r.updateConditions(remediationTimedOutByNhc, mdr)
+		updateRequired, err := r.updateConditions(remediationTimedOutByNhc, mdr)
+		if err == nil && updateRequired {
```

Comment:
for readability I would prefer this:
```
if updateRequired, err := ...; err != nil {
  return ctrl.Result{}, err
} else if updateRequired {
  ...
}
return ctrl.Result{}, nil
```
but I won't block on this... same at the other code place

Reply:
sgtm